File: academic-system-frontend/src/app/services/course.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Course } from '../models/course.model';
import { CourseCreateRequestDTO, CourseUpdateRequestDTO } from '../models/dtos/course.dto';

@Injectable({
  providedIn: 'root'
})
export class CourseService {
  private apiUrl = 'http://localhost:8080/api/courses';

  constructor(private http: HttpClient) { }

  getCourses(): Observable<Course[]> {
    return this.http.get<Course[]>(this.apiUrl);
  }

  createCourse(course: CourseCreateRequestDTO): Observable<Course> {
    return this.http.post<Course>(this.apiUrl, course);
  }

  updateCourse(id: string, course: CourseUpdateRequestDTO): Observable<Course> {
    return this.http.put<Course>(`${this.apiUrl}/${id}`, course);
  }

  deleteCourse(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: src/main/java/com/chrystian/sistemaacademicodematriculas/contract/CourseService.java
package com.chrystian.sistemaacademicodematriculas.contract;

import com.chrystian.sistemaacademicodematriculas.dto.CourseCreateRequestDTO;
import com.chrystian.sistemaacademicodematriculas.dto.CourseUpdateRequestDTO;
import com.chrystian.sistemaacademicodematriculas.model.Course;

import java.util.List;
import java.util.UUID;

public interface CourseService {
    List<Course> list();
    Course create(CourseCreateRequestDTO courseDTO);
    Course update(UUID id, CourseUpdateRequestDTO courseDTO);
    void delete(UUID id);
}